app/Http/Controllers/Rpc/WalletController.php

```diff
@@ -65,31 +72,36 @@ public function calculateWithdrawal(Request $request)
 
     public function withdraw(Request $request)
     {
+        $addressFrom = $this->getAdServerAdsAddress();
+        if (!AdsValidator::isAccountAddressValid($addressFrom)) {
+            Log::error("Invalid ADS address is set: ${addressFrom}");
+            return self::json([], Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+
         Validator::make($request->all(), [
             self::FIELD_AMOUNT => [self::VALIDATOR_RULE_REQUIRED, 'integer', 'min:1'],
             self::FIELD_TO => self::VALIDATOR_RULE_REQUIRED,
         ])->validate();
-
         $amount = $request->input(self::FIELD_AMOUNT);
         $addressTo = $request->input(self::FIELD_TO);
+        //TODO validate memo
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1135)